src/controllers/attachment/index.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  Security,
+  Tags,
+  UploadedFile,
+  Request,
+  Controller,
+  Get,
+  Route,
+  Path,
+  Post,
+  Header,
+  Response,
+  SuccessResponse,
+} from 'tsoa'
+import { Logger } from 'pino'
+import express from 'express'
+
+import { logger } from '../../lib/logger'
+import Database, { Models, Query } from '../../lib/db'
+import type { Attachment } from '../../models'
+import { BadRequest, NotFound } from '../../lib/error-handler'
+
+type File = {
+  [k: string]: string
+}
+
+@Route('attachment')
+@Tags('attachment')
+@Security('bearerAuth')
+export class attachment extends Controller {
+  log: Logger
+  dbClient: Database = new Database()
+  db: Models<Query> | any // TMP this is the only one could not address now
+
+  constructor() {
+    super()
+    this.log = logger.child({ controller: '/attachment' })
+    // TMP will be updated with a wrapper so ORM client independant
+    this.db = this.dbClient.db()
+  }
+
+  @Get('/')
+  @SuccessResponse(200, 'returns all attachments')
+  public async get(): Promise<Attachment[]> {
+    this.log.debug('retrieving all attachments')
+    const result = await this.db.attachment()
+
+    return result.map(({ binary_blob, ...item }: Attachment) => ({
+      ...item,
+      size: binary_blob.size,
+    }))
+  }
+
+  @Post('/')
+  @SuccessResponse(201, 'attachment has been created')
+  public async create(@Request() req: express.Request, @UploadedFile() file: File): Promise<string> {
+    this.log.debug(`creating an attachment ${JSON.stringify(file || req.body)}`)
+    if (file) {
+      const id = await this.db
+        .attachment()
+        .insert({
+          filename: file.originalname,
+          binary_blob: Buffer.from(file.buffer),
+        })
+        .returning('id')
+        .then((row: any) => row[0].id)
+
+      return `${id} binary attachment has been created`
+    }
+
+    if (!req.body) throw new BadRequest('nothing to upload')
+
+    const id = await this.db
+      .attachment()
+      .insert({
+        filename: 'json',
+        binary_blob: Buffer.from(JSON.stringify(req.body)),
+      })
+      .then((row: Array<unknown>) => row[0]) // return id implement along with db changes since it already addresses this
+
+    return `${id} JSON attachment has been created`
```

Comment:
also why are we just returning a string?

Reply:
presviously we returned an AttachmentEntry https://github.com/digicatapult/inteli-api/blob/7fe3fbc2edc68d3483fcfde251f6fd6b610048c3/app/api-v1/routes/attachment.js#L69